### ISSUE TITLE
Aggregate function in query causing issues during search (and yes, i used searchable: false)

### ISSUE THREAD

**Issue by dejan7** [yajra/laravel-datatables#616]
Hello, i have a query with aggregate function like this. Notice the Round(Avg(... thing in select

```
public function listAll()
{
    $users = DB::table('testtaker as tt')
        ->join('company as c1', 'tt.company_id', '=', 'c1.id')
        ->join('users as u2', 'tt.added_by', '=', 'u2.id')
        ->leftJoin('test_attempt as ta', 'tt.id', '=', 'ta.testtaker_id')
        ->groupBy('tt.id')
        ->whereNull('tt.deleted_at');
        ->select(['tt.id', 'tt.first_name', 'tt.last_name', 'tt.phone_number', 'tt.address', 'tt.email', 'c1.company_name', 'c1.logo', DB::raw('CONCAT(u2.first_Name, " ", u2.last_Name) AS added_by'), DB::raw('ROUND(AVG(ta.total_points_percentage)) AS avg_grade')]);

         return Datatables::of($users)->make(true);
}
```

And this is the JS code:

```
jQuery('#' + tableid).DataTable( {
            processing: true,
            serverSide: true,
            columns: [
             { data: 'id', name: 'tt.id'},
             { data: 'company_name', name: 'c1.company_name', bSearchable: false, searchable: false},
             { data: 'added_by', name: 'added_by', bSearchable: false, searchable: false},
             { data: 'first_name', name: 'tt.first_name'},
             { data: 'last_name', name: 'tt.last_name'},
             { data: 'avg_grade', name: 'avg_grade', bSearchable: false, searchable: false},
             { data: 'email', name: 'tt.email'},
             { data: 'phone_number', name: 'tt.phone_number'},
             { data: 'address', name: 'tt.address'}
         ],
            ......
```

On initial load everything **does work.**. However, when i try typing in the Datatables' default search box filter the following exception occurs on server:

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'round(avg(ta.total_points_percentage))' in 'where clause'' in /[location]/Connection.php:333

Next exception 'Illuminate\Database\QueryException' with message 'SQLSTATE[42S22]: Column not found: 1054 Unknown column 'round(avg(ta.total_points_percentage))' in 'where clause' (SQL: select count(*) as aggregate from (select `tt`.`id`, `tt`.`first_name`, `tt`.`last_name`, `tt`.`phone_number`, `tt`.`address`, `tt`.`email`, `c1`.`company_name`, `c1`.`logo`, CONCAT(u2.first_Name, " ", u2.last_Name) AS added_by, ROUND(AVG(ta.total_points_percentage)) AS avg_grade from `testtaker` as `tt` inner join `company` as `c1` on `tt`.`company_id` = `c1`.`id` inner join `users` as `u2` on `tt`.`added_by` = `u2`.`id` left join `test_attempt` as `ta` on `tt`.`id` = `ta`.`testtaker_id` where `tt`.`deleted_at` is null and `tt`.`deleted_at` is null and (LOWER(`tt`.`id`) LIKE %t% or LOWER(`tt`.`first_name`) LIKE %t% or LOWER(`tt`.`last_name`) LIKE %t% or LOWER(`tt`.`email`) LIKE %t% or LOWER(`tt`.`phone_number`) LIKE %t% or LOWER(`tt`.`address`) LIKE %t% or LOWER(`round(avg(ta`.`total_points_percentage))`) LIKE %t%) group by `tt`.`id`) count_row_table)' in /[location]/Connection.php:713

Basically it messes up the select columns during count query. 

Any ideas what could be causing this? Thanks a lot. 
- Operating System Ubuntu  (Laravel Homestead)
- PHP Version: 7.0
- Laravel Version: 5.2.34
- Laravel-Datatables Version: ~6.0

**Comment by yajra** [comment]
I think you need to use [filterColumn](http://datatables.yajrabox.com/eloquent/post-column-search) and write the correct query. 

However, it's weird why it's being search when it supposed to be skipped. Are you using the latest version?

**Comment by dejan7** [comment]
I was using v6.11.5. Tried updating to v6.13.1, but no luck.

I managed to fix it with this ugly hack:

        return Datatables::of($users)
            ->filterColumn('round(avg(ta.total_points_percentage))', function($query, $keyword) {

            })
            ->make(true);

because the error that was getting thrown was 
`Column not found: 1054 Unknown column 'round(avg(ta.total_points_percentage))' in 'where clause'`

So thanks for the tip.

But i'll try to investigate further and i will let you know what is causing this.

**Comment by swilla** [comment]
I'm getting a similar issue:

> Invalid parameter number (SQL: select count(*) as aggregate from (select '1' as `row_count` from `market_routes` where (LOWER(`market_routes`.`image_url`) LIKE %%d%% or (select count(1) from `markets` where `market_routes`.`market_id` = `markets`.`id` and LOWER(`city`) LIKE %D% and `markets`.`enabled` = %%d%%) >= 1 or LOWER(`market_routes`.`name`) LIKE %%d%% or LOWER(`market_routes`.`direction`) LIKE %%d%% or LOWER(`market_routes`.`speed_percent`) LIKE %%d%% or LOWER(`market_routes`.`updated_at`) LIKE ?) and `market_routes`.`deleted_at` is null) count_row_table)


```
public function getDatatables()
    {
        $marketRoutes = MarketRoute::with(['market'])
            ->select([
                'market_routes.id',
                'market_routes.market_id',
                'market_routes.name',
                'market_routes.image_url',
                'market_routes.direction',
                'market_routes.speed_percent',
                'market_routes.updated_at',
            ]);

        return Datatables::of($marketRoutes)
            ->editColumn('image_url', function ($route) {
                if ($route->image_url) {
                    return '<img src="'.$route->image_url.'" style="max-width: 100px; height: auto" />';
                }
                return '';
            })
            ->addColumn('action', function ($route) {
                return '<a href="'.route("admin.market-routes.view", [$route->id]).'" class="btn btn-xs btn-primary"><i class="fa fa-eye"></i></a>';
            })
            ->make(true);
    }
```

```
var template = Handlebars.compile($("#details-template").html());
    var table = $('#market-routes').DataTable({
        processing: true,
        serverSide: true,
        ajax: '{!! route("admin.market-routes.datatables") !!}',
        columns: [
            {
                "className": 'details-control',
                "orderable": false,
                "searchable": false,
                "data": null,
                "defaultContent": '<a href="#">+</a>'
            },
            {data: 'id', name: 'market_routes.id', searchable: false, orderable: false},
            {data: 'image_url', name: 'market_routes.image_url'},
            {data: 'market.city', name: 'market.city'},
            {data: 'name', name: 'market_routes.name'},
            {data: 'direction', name: 'market_routes.direction'},
            {data: 'speed_percent', name: 'market_routes.speed_percent'},
            {data: 'updated_at', name: 'market_routes.updated_at'},
            {data: 'action', name: 'action', orderable: false, searchable: false}
        ]
    });

    // Add event listener for opening and closing details
    $('tbody').on('click', 'td.details-control', function () {
        var tr = $(this).closest('tr');
        var row = table.row( tr );

        if ( row.child.isShown() ) {
            // This row is already open - close it
            row.child.hide();
            tr.removeClass('shown');
        }
        else {
            // Open this row
            row.child( template(row.data()) ).show();
            tr.addClass('shown');
        }

        return false;
    });
```

**Comment by swilla** [comment]
It appears that this example might have the same issue: 
http://datatables.yajrabox.com/eloquent/row-details

**Comment by yajra** [comment]
Thanks for reporting. The error on demo was due to missing searchable flag on row details and is now fixed. But it does not have any custom sql like `avg` which makes it not related to this issue.

```php
{
    "className":      'details-control',
    "orderable":      false,
    "searchable":     false,
    "data":           null,
    "defaultContent": ''
},
```

**Comment by yajra** [comment]
Closing old questions. Please open a new one if still having issues. Thanks!
